task21/js/main.js

```diff
@@ -0,0 +1,154 @@
+    addEventForSortButton(userContentsData);
+}
+initialize();
+
+const userTableColumn =  {
+    "id" : "ID",
+    "name" : "名前",
+    "gender" : "性別",
+    "age" : "年齢"
+};
+
+const renderTableContents = (userTableColumn, userContentsData) => {
+    const tableElement = document.createElement("table");
+
+    tableElement.classList.add("mt-20", "mx-auto", "w-3/5");
+
+    parent.appendChild(tableElement).appendChild(getCreatedTableHeader(userTableColumn)).after(getCreatedTableBody(userContentsData));
+};
+
+const getCreatedTableHeader = (userTableColumn) => {
+    const theadElement = document.createElement("thead");
+    const tableRowElement = document.createElement("tr");
+    const tableHeaderFragment = document.createDocumentFragment();
+    for(const column of Object.values(userTableColumn)){
+        const tableHeaderElement = createElementWithClassName("th", "table-header");
+        tableHeaderElement.classList.add("bg-gray-900", "text-white");
+        tableHeaderElement.textContent = column;
+        column === "ID" && tableHeaderElement.appendChild(getCreatedSortButton());
+        tableHeaderFragment.appendChild(tableHeaderElement);
+    }
+    theadElement.appendChild(tableRowElement).appendChild(tableHeaderFragment);
+    return theadElement;
+}
+
+const getCreatedTableBody = (userContentsData) => {
+    const tbodyElement = document.createElement("tbody");
+    tbodyElement.id = "js-tableBody";
+    const tableBodyFragment = document.createDocumentFragment();
+    for(const user of userContentsData){
+        const tableRowElement = document.createElement("tr");
+        for(const column of Object.keys(userTableColumn)){
+            const tableDataElement = createElementWithClassName("td", "table-data");
+            tableDataElement.textContent = user[column];
+            tableBodyFragment.appendChild(tableRowElement).appendChild(tableDataElement);
+        }
+    }
+    tbodyElement.appendChild(tableBodyFragment);
+    return tbodyElement;
+};
+
+const getCreatedSortButton = () => {
+    const sortButton = document.createElement("button");
+    const sortImage = createElementWithClassName("img", "sort-image");
+    sortButton.id = "js-sortButton";
+    sortImage.dataset.status = "default";
+    sortButton.appendChild(sortImage);
+    return sortButton;
+};
+
+const addEventForSortButton = (userContentsData) => {
+    const sortButton = document.getElementById("js-sortButton");
+    sortButton.addEventListener('click', (e) => {
+        const nextStatus = changeSortStatus(e.target.dataset.status);
+        e.target.dataset.status = nextStatus;
+
+        sortContetns(userContentsData, nextStatus);        
+    })
+}
+
+const changeSortStatus = (status) => {
+    switch(status){
+        case "default":
+            return "asc";
+        case "asc":
+            return "desc";
+        case "desc":
+            return "default";
+        default:
+            return "default";
+    }
```

Comment:
```suggestion
        case "desc":
        default:
            return "default";
    }
```

Reply:
Fixed it.Thank you.
[3535796](https://github.com/seeeen044/JavaScript_handson/pull/22/commits/35357965c8c32dcde7d5531314069fe7f90f7a7a)